src/guards/attendance-time-slot-group-admin.guard.ts

```diff
@@ -0,0 +1,49 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { TimeSlotRepository } from '../repositories/time-slot.repository';
+import { GroupRepository } from '../repositories/group.repository';
+import { AttendanceRepository } from 'src/repositories/attendance.repository';
+
+@Injectable()
+export class AttendanceTimeSlotGroupAdminGuard implements CanActivate {
+  constructor(
+    private readonly groupRepository: GroupRepository,
+    private readonly timeSlotRepository: TimeSlotRepository,
+    private readonly attendanceRepository: AttendanceRepository,
+  ) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest();
+
+    const attendance = await this.attendanceRepository.findOneById(
+      request.params.attendanceId,
+    );
+
+    const timeSlot = await this.timeSlotRepository.findOneById(
+      // @ts-ignore
```

Comment:
is it necessary ?

Reply:
yes it is .. Because of the magic of typescript